### PR TITLE
[CI] Using mmg 5.5

### PR DESCRIPTION
**Pull request by marcnunezc**
**Description**
Switches to MMG 5.5 in tests. MMG 5.4 will be readded after changing the docker image compilation. Also preparing the ground to use 5.5 on one compiler and 5.4 in the other as suggested by @philbucher  

**Changelog**
- Switch to mmg 5.5 in tests.

### CODE REVIEW

**Review by philbucher — DISMISSED** [review]

**Comment by marcnunezc** [comment]
Hmm it failed. Is the docker image available immediately after merging #7767? Will check what happens with env variables locally.

**Comment by philbucher** [comment]
> Hmm it failed. Is the docker image available immediately after merging #7767? Will check what happens with env variables locally.

well it still needs to upload it to dockerhub but that should be fast
If it still fails after restarting then sth is off

**Comment by marcnunezc** [comment]
Checking the docker, the libs are correclty written in /external_libraries/mmg/mmg_5_5_1/lib , also the ENV variables.

image

image

If it fails again, my only guess is that the libs at /usr/local/lib are messing around. 
image

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -263,6 +359,10 @@ jobs:
         . /opt/intel/mkl/bin/mklvars.sh
         export PYTHONPATH=${PYTHONPATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}
         export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:${GITHUB_WORKSPACE}/bin/${{ matrix.build-type }}/libs
+        if [ ${{ matrix.compiler }} = gcc ]; then
+          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/external_libraries/mmg/mmg_5_5_1/lib
+        elif [ ${{ matrix.compiler }} = clang ]; then
+          export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/external_libraries/mmg/mmg_5_4_1/lib
```
Comment (philbucher) [inline .github/workflows/ci.yml]: hm this grows more and more out of hand...
did you try adding both to the path (in this PR?)

I will think of a better solution for the future, sth like an env file for the docker

Comment (marcnunezc) [reply .github/workflows/ci.yml]: Yep, not a fan of this solution either.

Let's check if it works fine like this and then I will try adding both.

**Comment by marcnunezc** [comment]
v5.5 tests are failing, but in local they work fine :( 

No idea what is the issue

**Comment by philbucher** [comment]
maybe 5.5 just gives different results?

@loumalouomega any idea?

**Comment by loumalouomega** [comment]
> maybe 5.5 just gives different results?
> 
> @loumalouomega any idea?

I already patched that, in theory

**Comment by loumalouomega** [comment]
I updated the results for 5.5, not 5.5.1...

**Comment by marcnunezc** [comment]
> > maybe 5.5 just gives different results?
> > @loumalouomega any idea?
> 
> I already patched that, in theory

Yeah, in my machine everything works fine.

Right now im compiling inside the docker in local, so I can debug it there

**Comment by marcnunezc** [comment]
> I updated the results for 5.5, not 5.5.1...

🤔 , I tested 5.5.1 in my machine in any case :/

**Comment by marcnunezc** [comment]
I just found the issue! In v5.5, SCOTCH is enabled by default (which is a library that reorders the output mesh to improve its quality), which breaks our tests (the order of the output nodes is different).

**Comment by marcnunezc** [comment]
Closing this one as it is a bit dirty, see #7781. Will open a new PR adding again INCLUDE_MMG with the correct paths

**Pull request closed without merging**